src/TruePBR.cpp

```diff
@@ -0,0 +1,1496 @@
+
+				grassProperty->SetFlags(RE::BSShaderProperty::EShaderPropertyFlag8::kMenuScreen, true);
+
+				auto pbrMaterial = static_cast<BSLightingShaderMaterialPBR*>(grassProperty->material);
+				pbrMaterial->pbrFlags = pbrSrcMaterial->pbrFlags;
+				pbrMaterial->normalTexture = pbrSrcMaterial->normalTexture;
+				pbrMaterial->rmaosTexture = pbrSrcMaterial->rmaosTexture;
+				pbrMaterial->featuresTexture0 = pbrSrcMaterial->featuresTexture0;
+				pbrMaterial->featuresTexture1 = pbrSrcMaterial->featuresTexture1;
+				pbrMaterial->specularColorScale = pbrSrcMaterial->specularColorScale;
+				pbrMaterial->specularPower = pbrSrcMaterial->specularPower;
+				pbrMaterial->specularColor = pbrSrcMaterial->specularColor;
+				pbrMaterial->subSurfaceLightRolloff = pbrSrcMaterial->subSurfaceLightRolloff;
+				pbrMaterial->coatRoughness = pbrSrcMaterial->coatRoughness;
+			}
+		}
+
+		return grassProperty;
+	}
+	static inline REL::Relocation<decltype(thunk)> func;
+};
+
+struct BSGrassShaderProperty_GetRenderPasses
+{
+	static RE::BSShaderProperty::RenderPassArray* thunk(RE::BSLightingShaderProperty* property, RE::BSGeometry* geometry, std::uint32_t renderFlags, RE::BSShaderAccumulator* accumulator)
+	{
+		auto renderPasses = func(property, geometry, renderFlags, accumulator);
+		if (renderPasses == nullptr) {
+			return renderPasses;
+		}
+
+		const bool isPbr = property->flags.any(RE::BSShaderProperty::EShaderPropertyFlag::kMenuScreen);
+		if (isPbr) {
+			auto currentPass = renderPasses->head;
+			while (currentPass != nullptr) {
+				if (currentPass->shader->shaderType == RE::BSShader::Type::Grass && currentPass->passEnum != 0x5C00005C) {
+					currentPass->passEnum = 0x5C000042;
+				}
+				currentPass = currentPass->next;
+			}
+		}
+
+		return renderPasses;
+	}
+	static inline REL::Relocation<decltype(thunk)> func;
+};
+
+struct BSGrassShader_SetupTechnique
+{
+	static bool thunk(RE::BSShader* shader, uint32_t globalTechnique)
+	{
+		if (globalTechnique == 0x5C000042) {
+			auto shadowState = RE::BSGraphics::RendererShadowState::GetSingleton();
+			auto* graphicsState = RE::BSGraphics::State::GetSingleton();
+			auto* renderer = RE::BSGraphics::Renderer::GetSingleton();
+
+			const uint32_t localTechnique = static_cast<uint32_t>(SIE::ShaderCache::GrassShaderTechniques::TruePbr);
+			uint32_t shaderDescriptor = localTechnique;
+			if (graphicsState->useEarlyZ) {
+				shaderDescriptor |= static_cast<uint32_t>(SIE::ShaderCache::GrassShaderFlags::AlphaTest);
+			}
+
+			const bool began = Hooks::hk_BSShader_BeginTechnique(shader, shaderDescriptor, shaderDescriptor, false);
+			if (!began) {
+				return false;
+			}
+
+			static auto fogMethod = REL::Relocation<void (*)()>(REL::RelocationID(100000, 106707));
+			fogMethod();
+
+			static auto* bShadowsOnGrass = RE::GetINISetting("bShadowsOnGrass:Display");
+			if (!bShadowsOnGrass->GetBool()) {
+				shadowState->SetPSTexture(1, graphicsState->defaultTextureWhite->rendererTexture);
```

Comment:
`graphicsState->defaultTextureWhite` -> `graphicsState->GetRuntimeData().defaultTextureWhite`

Makes it work in VR and AE later than 1.6.629

Reply:
Fixed